libraries/Microsoft.Bot.Connector/Authentication/PasswordServiceClientCredentialFactory.cs

```diff
@@ -85,30 +108,21 @@ public override Task<ServiceClientCredentials> CreateCredentialsAsync(string app
 
             if (loginEndpoint.StartsWith(AuthenticationConstants.ToChannelFromBotLoginUrlTemplate, StringComparison.OrdinalIgnoreCase))
             {
-                return Task.FromResult<ServiceClientCredentials>(
-                    AppId == null
-                        ?
-                    MicrosoftAppCredentials.Empty
-                        :
-                    new MicrosoftAppCredentials(appId, Password, _httpClient, _logger, oauthScope));
+                return Task.FromResult<ServiceClientCredentials>(string.IsNullOrWhiteSpace(TenantId)
+                    ? new MicrosoftAppCredentials(appId, Password, _httpClient, _logger, oauthScope)
+                    : new MicrosoftAppCredentials(appId, Password, TenantId, _httpClient, _logger, oauthScope));
             }
             else if (loginEndpoint.Equals(GovernmentAuthenticationConstants.ToChannelFromBotLoginUrl, StringComparison.OrdinalIgnoreCase))
             {
-                return Task.FromResult<ServiceClientCredentials>(
-                    AppId == null
-                        ?
-                    MicrosoftGovernmentAppCredentials.Empty
-                        :
-                    new MicrosoftGovernmentAppCredentials(appId, Password, _httpClient, _logger, oauthScope));
+                return Task.FromResult<ServiceClientCredentials>(string.IsNullOrWhiteSpace(TenantId)
```

Comment:
see comment above - can this be simplified?

Reply:
Simplified the code.